### DISCUSSION TITLE
Broker didn't use bearer authentication

### DISCUSSION THREAD

**Discussion by nxtpl** [jasny/sso#129] [Help needed] [answered]
Hello,
I have set up the server and I've put the broker demo on another domain.

When opening the /broker/index.php, the login form is hidden and there is a red message "Broker didn't use bearer authentication" - is it something with my code? Similar thing happens with the AJAX demo.

UPDATE:
It seems to be connected with the unavailable $_SERVER['HTTP_AUTHORIZATION'] variable - how to fix it?

UPDATE 2 (resolved!):
It seems that on certain Apache servers, the header is being stripped for _security reasons_.
The problem is resolved by adding the lines below to my .htaccess file of the "server part" installation dir.

`RewriteEngine on`
`RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization},L]`

**Comment by jasny** [comment, marked as answer]
https://stackoverflow.com/questions/26549250/apache-strips-down-authorization-header

> Several modules will strip the Authorization header to try to enhance security by preventing scripts from seeing usernames, passwords, etc... unless the developer explicitly enables this.  Many of these modules will allow this header if you simply add the following line to .htaccess: `CGIPassAuth on` (source: [Apache docs](https://httpd.apache.org/docs/2.4/en/mod/core.html#cgipassauth) and [StackOverflow](https://stackoverflow.com/questions/17018586/apache-2-4-php-fpm-and-authorization-headers))

**Comment by jimmySkims** [comment]
Thanks for the answer! Great info!
